lib/include/jxl/encode.h

```diff
@@ -110,127 +110,143 @@ typedef enum {
    * specifically set it.
    */
-  JXL_ENC_OPTION_GROUP_ORDER_CENTER_X = 13,
+  JXL_ENC_OPTION_GROUP_ORDER_CENTER_X = 14,
 
   /** Determines the center for the center-first group order. Use -1 to
    * automatically use the middle of the image, 0..ysize to specifically set it.
    */
-  JXL_ENC_OPTION_GROUP_ORDER_CENTER_Y = 14,
+  JXL_ENC_OPTION_GROUP_ORDER_CENTER_Y = 15,
 
   /** Enables or disables progressive encoding for modular mode. Use -1 for the
    * encoder default, 0 to disable, 1 to enable.
    */
-  JXL_ENC_OPTION_RESPONSIVE = 15,
+  JXL_ENC_OPTION_RESPONSIVE = 16,
 
   /** Set the progressive mode for the AC coefficients of VarDCT, using spectral
    * progression from the DCT coefficients. Use -1 for the encoder default, 0 to
    * disable, 1 to enable.
    */
-  JXL_ENC_OPTION_PROGRESSIVE_AC = 16,
+  JXL_ENC_OPTION_PROGRESSIVE_AC = 17,
 
   /** Set the progressive mode for the AC coefficients of VarDCT, using
    * quantization of the least significant bits. Use -1 for the encoder default,
    * 0 to disable, 1 to enable.
    */
-  JXL_ENC_OPTION_QPROGRESSIVE_AC = 17,
+  JXL_ENC_OPTION_QPROGRESSIVE_AC = 18,
 
   /** Set the progressive mode using lower-resolution DC images for VarDCT. Use
    * -1 for the encoder default, 0 to disable, 1 to have an extra 64x64 lower
    * resolution pass, 2 to have a 512x512 and 64x64 lower resolution pass.
    */
-  JXL_ENC_OPTION_PROGRESSIVE_DC = 18,
+  JXL_ENC_OPTION_PROGRESSIVE_DC = 19,
 
   /** Use Global channel palette if the amount of colors is smaller than this
    * percentage of range. Use 0-100 to set an explicit percentage, -1 to use the
    * encoder default. Used for modular encoding.
    */
-  JXL_ENC_OPTION_CHANNEL_COLORS_PRE_TRANSFORM_PERCENT = 19,
+  JXL_ENC_OPTION_CHANNEL_COLORS_PRE_TRANSFORM_PERCENT = 20,
 
   /** Use Local channel palette if the amount of colors is smaller than this
    * percentage of range. Use 0-100 to set an explicit percentage, -1 to use the
    * encoder default. Used for modular encoding.
    */
-  JXL_ENC_OPTION_CHANNEL_COLORS_PERCENT = 20,
+  JXL_ENC_OPTION_CHANNEL_COLORS_PERCENT = 21,
 
   /** Use color palette if amount of colors is smaller than or equal to this
    * amount, or -1 to use the encoder default. Used for modular encoding.
    */
-  JXL_ENC_OPTION_PALETTE_COLORS = 21,
+  JXL_ENC_OPTION_PALETTE_COLORS = 22,
 
   /** Enables or disables delta palette. Use -1 for the default (encoder
    * chooses), 0 to disable, 1 to enable. Used in modular mode.
    */
-  JXL_ENC_OPTION_LOSSY_PALETTE = 22,
+  JXL_ENC_OPTION_LOSSY_PALETTE = 23,
+
+  /** Color transform for internal encoding: -1 = default, 0=XYB, 1=none (RGB),
+   * 2=YCbCr.
+   */
+  JXL_ENC_OPTION_COLOR_TRANSFORM = 24,
 
   /** Color space for modular encoding: 0=RGB, 1=YCoCg, 2-37=RCT, -1=default:
    * try several, depending on speed.
    */
-  JXL_ENC_OPTION_MODULAR_COLOR_SPACE = 23,
+  JXL_ENC_OPTION_MODULAR_COLOR_SPACE = 25,
```

Comment:
This is how it is in cjxl, but those values are a bit of a historically evolved mess. Basically 0 and 1 are special values and 2+n gets mapped to spec-RCT n. This is redundant because 0 is the same as n=0 and 1 is the same as n=6.

Before we put this in the API, maybe it is best to sanitize this and get rid of the special values 0 and 1. For cjxl we can see if we want to keep the old behavior or also sanitize it (it's an option you should rarely set except maybe to force it to 0, but for 0 the new saner behavior would be identical to the old behavior).

Reply:
I've updated the meaning of the API to match the spec. So the cjxl 0 and 1 can now be reached using 0 and 6. It's best if the API follows the specification, so with this updated comment and implementation it should match. It converts it to the internal C++ codec representation internally now.